carta/cpp/CartaLib/AxisDisplayInfo.h

```diff
@@ -0,0 +1,134 @@
+/***
+ * Information about how an axis should be displayed.
+ */
+
```

Comment:
I guess most of the work I have been doing for the past month will need 
to be redone then,
as I was apparently not following the official approach.  Since the 
method was not provided,
I assumed you were moving away from the CASA approach of transposing the 
CoordinateSystem.

Instead of adding a method to ImageInterface and then passing around the 
transposed
image to the necessary class, they are being passed permute information.
AxisDisplayInfo is the axis information needed by the grid renderer to 
display the axis when it is permuted.

On 10/16/2015 02:26 PM, Pavol Federl wrote:
>
> In carta/cpp/CartaLib/AxisDisplayInfo.h 
> <https://github.com/Astroua/CARTAvis/pull/88#discussion_r42286187>:
>
> > @@ -0,0 +1,134 @@
> > +/***
> > + * Information about how an axis should be displayed.
> > + */
> > +
>
> The purpose of this class is not clear to me from the comment. By 
> glancing at the code, it seems to be used in places where axes need to 
> be permuted in an ImageInterface...
>
> For axis permutations I was thinking of implementing something 
> generic, e.g. by adding a method to ImageInterface:
>
> virtual Image::ImageInterface * permuted(
> const std::vector & indices)
>
> which would return a special instance of ImageInterface. This way you 
> could permute axes and/or even remove some, without having to insert 
> permutation specific code anywhere else...
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Astroua/CARTAvis/pull/88/files#r42286187>.Web Bug 
> from 
> https://github.com/notifications/beacon/AGSu-1oOmwiFri9QvjQPrHzJa3pfmSjoks5o8VT6gaJpZM4GOLpe.gif
>